pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -228,13 +231,23 @@ async def _initialize_context(self):
     def _register_to_signal(self):
         on_termination_signal = functools.partial(self._on_termination_signal, Constants.termination_signal.name)
         on_drain_signal = functools.partial(self._on_drain_signal, Constants.drain_signal.name)
+        on_continue_signal = functools.partial(self._on_continue_signal, Constants.continue_signal.name)
 
         asyncio.get_running_loop().add_signal_handler(Constants.termination_signal, on_termination_signal)
         asyncio.get_running_loop().add_signal_handler(Constants.drain_signal, on_drain_signal)
+        asyncio.get_running_loop().add_signal_handler(Constants.continue_signal, on_continue_signal)
 
     def _on_drain_signal(self, signal_name):
+        # do not perform draining if discarding events
+        if self._discard_events:
+            return
+
         self._logger.debug_with('Received signal, calling draining callback', signal=signal_name)
```

Comment:
We no longer call if from here, can be removed. (Same for termination on line 256)
```suggestion
        self._logger.debug_with('Received signal', signal=signal_name)
```